### ISSUE TITLE
Serial::write(String string) throwing java.lang.StringIndexOutOfBoundsException

### ISSUE THREAD

**Issue by prsebastian** [Pi4J/pi4j-v1#76]
In some situations, the above-mentioned method is throwing a  StringIndexOutOfBoundsException. 

Example: 

```
serial.write( ":VAL\u0000VM\u0000\u0001\u0000\u0001A\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FFVariableR\n\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00FF\u00D0" );
```

This is by the way the same bug as in Issue 37: https://github.com/Pi4J/pi4j/issues/37
which you didn't really fix :) but rather postponed it by not calling write(new String(data)); anymore, 
as seen in https://github.com/Pi4J/pi4j/commit/8c0ebb59259fbed90c0c18fbd306a463b332f648
but the bug itself is not in write(byte[] bytes) but in write(String string).

best regards,
Sebastian.

**Comment by savageautomate** [comment]
I have been working on a new serial implementation to properly support a byte level interface for working with the data.  I suggest that you try the experimental build 1.1-SNAPSHOT.  The serial API is completely overhauled and should provide far better results when working with bytes. 

GITHUB BRANCH:
https://github.com/Pi4J/pi4j/tree/serial_enhancements

DOWNLOAD:
http://get.pi4j.com/download/pi4j-1.1-SNAPSHOT.zip

Please let me know if you find any problems ... remember it is very new and experimental :-)

**Comment by savageautomate** [comment]
Fixed in 1.1-SNAPSHOT build
